examples/slide_puzzle/main.rs

```diff
@@ -231,3 +235,43 @@ pub fn main() {
     });
     main_window.run().unwrap();
 }
+
+#[cfg(target_arch = "wasm32")]
+/// winit doesn't handle the resizing of the canvas from CSS well
+/// https://github.com/rust-windowing/winit/issues/1661
+/// in the mean time, adjust the size manually
+fn handle_resize(slint_window: slint::Weak<MainWindow>) -> Option<()> {
+    let window = web_sys::window()?;
+    let closure = move || {
+        let window = web_sys::window()?;
+        let doc = window.document()?;
+
+        let container = doc.get_element_by_id("container")?;
+        let children = container.children();
+        let mut height_sum = 0.;
+        for i in 0..children.length() {
+            let child = children.item(i).unwrap();
+            if child.tag_name().to_lowercase() == "canvas" {
+                continue;
+            }
+            let style = window.get_computed_style(&child).ok()?;
+            let get_margin = |m| {
+                style.as_ref()?.get_property_value(m).ok()?.strip_suffix("px")?.parse::<f32>().ok()
```

Comment:
```suggestion
            let get_margin = |m| {
                let style = window.get_computed_style(&child).ok()?;
                style.as_ref()?.get_property_value(m).ok()?.strip_suffix("px")?.parse::<f32>().ok()
```

Reply:
Why do you want to recompute the style twice ?

Reply:
Something like that would almost compile.  But get_computed_style returns a `Result<Option<_>, _>` 
I think we get an Err only if the get_computed_style throws.  I don't know under which condition this happens, but if it does something's wrong and it feels like aborting the resizing is not such a bad idea.
If it does return Ok(None), then we will consider there was no margins.